### ISSUE TITLE
Not displaying GroundOverlay in KML file.

### ISSUE THREAD

**Issue by dwhipps** [AnalyticalGraphicsInc/cesium#4212]
This is likely a two-part ticket, but for starters, the KML file I have here:

https://earthdata.layeredearth.com/Layers/Earth/Climate%20Change/Other/Malaria%20in%20Africa%20Projections.kmz
... does not display anything in Cesium, seen in SANDCASTLE here:

http://cesiumjs.org/Cesium/Apps/Sandcastle/?src=Hello%20World.html&label=Showcases&gist=a2c8d3fa055a3a341114040331265ac7

I suspect this has to do with the nested <Folder> elements in the kml, but I can see that processGroundOverlay is getting called, they're just not being displayed.

(The second part of this, which I'll put in a separate ticket, is that Cesium shouldn't be processing feature nodes unless they're 'visible' as defined here: https://developers.google.com/kml/documentation/kmlreference#folder)

> A Folder is used to arrange other Features hierarchically (Folders, Placemarks, NetworkLinks, or Overlays). A Feature is visible only if it and all its ancestors are visible.

**Comment by hpinkos** [comment]
Thanks @dwhipps!
I think the issue might be related to a bug we've seen with timespans: #4041

For visibility, from looking at the code it looks like that is being processed correctly.  When is sees visibility of zero, it sets show to false.  I think we still want to process it in case a user wants to set show to true later on.

**Comment by dwhipps** [comment]
It's not clear on whether this has a decent chance of getting fixed. It's a bit of a show-stopper, as nothing at all renders.

**Comment by hpinkos** [comment]
Sorry, I can't promise that we'll be able to get to it immediately.
We are planning to spend a few days in October tackling frequently requested bug fixes.  This seems like it might be a good issue for us to look into then since it is effecting multiple users.

**Comment by mramato** [comment]
@dwhipps thanks for the report.  If you're up for it, there's no need to "wait for us" to do something, Cesium is a community with lots of contributors and we are more than happy to take a pull request for this or any other issue.  This sounds like something that would be isolated to code in `KmlDataSource.js`

**Comment by dwhipps** [comment]
After further investigation, this appears to be a case of Google Earth (and WolrdWind) being more lenient than Cesium. There were multiple GroundOverlay nodes with the same node "id" (these should be unique), so when `getOrCreateEntity` is called inside `processFeature`, the existing entities were being found with that id, no new entity was created, and entity.show was being set to false (because the last instance was not 'visible'.)

I confirmed this by simply changing the duplicated node ids. All was fine after that.

In short, I don't think this is actually a bug in Cesium. Not sure if there's a way to warn if multiple nodes are found with the same ID, but seems like a real edge-case.

I'm closing this issue.
